File: 7kyu 2/insurance.js
/*
|--------------------------------------------------------------------------
| Cost of my ride
|--------------------------------------------------------------------------
|
*/

function insurance(age, size, numofdays) {
   let res = 0;
   res += (age < 25) ? numofdays * 10 : a = 0;
   res += size == "economy" ? 0 : size == "medium" ? numofdays * 10 : size == "full-size" ? numofdays * 15 : numofdays * 15;
   res += numofdays > 0 ? 50 * numofdays : 0;
   return res > 0 ? res : 0;
}


const insurance = (a, s, n) => 0 < n ? (50 + (a < 25) * 10 + (s == 'economy' ? 0 : s == 'medium' ? 10 : 15)) * n : 0;


function insurance(age, size, numofdays) {
   const extraAge = age < 25 ? 10 : 0;
   const extraSize = size == "economy" ? 0 : size == "medium" ? 10 : 15;
   return numofdays > 0 ? (50 + extraAge + extraSize) * numofdays : 0;
}


function insurance(age, size, numofdays) {
   return (50 + 10 * (age < 25) + (size === 'economy' ? 0 : size === 'medium' ? 10 : 15)) * Math.max(numofdays, 0);
}


function insurance(age, size, numofdays) {
   const surcharge =
      size === 'economy' ? 0 :
      size === 'medium' ? 10 : 15;
   return Math.max(numofdays, 0) * ((age < 25 ? 10 : 0) + 50 + surcharge);
}


console.log(insurance(18, "medium", 7), 490);
console.log(insurance(30, "full-size", 30), 1950);
console.log(insurance(21, "economy", -10), 0);
console.log(insurance(42, "my custom car", 7), 455);

File: Python/7py 1/insurance.py
"""
|--------------------------------------------------------------------------
| Cost of my ride
|--------------------------------------------------------------------------
|
"""

def insurance(age, size, num_of_days): 
   xAge = 10 if age < 25 else 0
   xSize = 0 if size == 'economy' else 10 if size == 'medium' else 15
   return (50 + xAge + xSize) * num_of_days if num_of_days > 0 else 0


insurance = lambda a, sz, n: (p:=n*((a<25)*10 + 50 + (sz!='economy')*(((sz=='medium') and 10) or 15)))>0 and p or 0


sc={'economy':0,'medium':10,'full-size':15}
def insurance(a,s,n): 
   return (50+(a<25)*10+sc.get(s,15))*max(0,n)


def insurance(age, size, num_of_days): 
   return (50 + (10 if age < 25 else 0) + (10 if size == "medium" else 0 if size == "economy" else 15)) * max(num_of_days, 0)


print(insurance(18, "medium", 7), 490)
print(insurance(30,"full-size",30),1950)
print(insurance(21,"economy",-10), 0)
print(insurance(42,"my custom car",7), 455)